Develop

inzanez opened the pull request.

Commit:
Switch to 'develop' of arangors

Commit:
Remove 'Normal' connection type, as all is the same now

Commit:
Adjust to use version 0.3 of blocking arangors

Commit:
Finalize build

Commit:
Add gitignore

Commit:
Change dependency version requirement

Commit:
Update to latest ArangoDB verison

ManevilleF commented on Cargo.toml:
```diff
@@ -1,15 +1,12 @@
 [dependencies]
-arangors = { git="https://github.com/inzanez/arangors", branch = "master" }
-failure = "0.1.6"
+arangors = { version = ">=0.4.3", features = ["reqwest_blocking"], default-features = false }
```
Hello,
Why use the reqwest blocking feature?

inzanez replied:
Hello,
I think the main reason would be that r2d2 is synchronous. But now you mention it, I'm not quite sure if that's a valid point. As the resulting database connection could still be used in an async way, even if the connection pool itself is not async.

If that's the case, I would introduce features to let the user choose, and maybe default to async.

ManevilleF replied:
Since I'm developping an object mapper for ArangoDB based on arangors ([aragog](https://crates.io/crates/aragog)) I'm interested in the question, right now I use the default features and have to use stuff like `async_trait` to use db operations.

But I'm wondering if I should use `reqwest_blocking` and put everything in a synchronous way: [Pull request](https://gitlab.com/qonfucius/aragog/-/merge_requests/10)

I'm also wondering if I should use a `r2d2` implementation or your crate directly..

### Edit:

I will look up on how is arangors really built for the http client and see what I should do, but if I want to implement r2d2 most definetly syncrhonous is required

inzanez replied:
Well, `r2d2` is the base of the connection pooling system. All the other crates you find like `r2d2-influx_db_client`, `r2d2-mongodb` or `r2d2-arangors` are here to enable `r2d2` to work with these specific databases. So it's more like an extension to `r2d2`, not a re-implementation. So if you were to use `r2d2` with `arangors` driver, you would use `r2d2-arangors` as a connection between the two.

Regarding sync or async, I think the `arangors` crate shows quite nicely how to write everything async and have it transformed to sync if required (using `maybe_async`). So as `async/await` is stable now, I would definitely go with `async` if you can.

I will initiate changes to `r2d2` in the meantime so that it will default to `async`. If that should not work out for whatever reason, there still `mobc` for full async connection pooling.

inzanez replied:
I came to the conclusion that I will leave `r2d2-arangors` on `sync`. Better use `mobc` and `mobc-arangors` for async, as this will make the connection pool itself `async` too.

arn-the-long-beard replied:
Hello guys :smile: 
Thank you for the PR.

Well, I use a lot **arangors** and I use it directly inside my **actix-web**  without any other dependencies.

I think that limiting dependencies is always good. So I am not sure if we need **r2d2** . Maybe I do not understand much about connection pooling system as well so.

I also use the **async** part always. Isn't it the recommended way to use **async** for web/db stuff for better performance ?

arn-the-long-beard replied:
Please tell me if you need help with something. Maybe I can give you a hand :smile: .

ManevilleF replied:
> Hello guys 😄
> Thank you for the PR.
> 
> Well, I use a lot **arangors** and I use it directly inside my **actix-web** without any other dependencies.
> 
> I think that limiting dependencies is always good. So I am not sure if we need **r2d2** . Maybe I do not understand much about connection pooling system as well so.
> 
> I also use the **async** part always. Isn't it the recommended way to use **async** for web/db stuff for better performance ?

Hello, I see you use a forked version of `arangors`, this PR is for `r2d2-arangors` Or maybe I misunderstood.

I use arangors with actix too, and to do so I made a Object modelers for `arangors` called `aragog` and I use that for my REST actix API... But I don't know if I should change my all `async` functions and methods into synchronous or `maybe_async` , @inzanez suggests I leave the async so I guess I won't put anything in synchronous

arn-the-long-beard replied:
I have a forked version of `arangors` because I help coding it sometimes :smile:  but I use the normal one :)

Okay I understand more now :+1:

I just realized we are on the `r2d2-arangors` and not  on `arangors` repos here. I must apologize for my spam! :stuck_out_tongue_closed_eyes:

inzanez replied:
Not an issue,...regarding the 'direct' usage of arangors in a web-application: I think that using connections pools gives you quite an advantage, as the pool (like `r2d2` or `mobc`) keeps a number of connections open and ready, so it can serve you a database connection very quickly (for instance directly in your web api endpoints). If you share a single connection between all the endpoints, that's not really performing well. I know that `actix` has other means to get around that, but I would still have a look into connection pooling. I think it's a better alternative than a separately running `actor`.

On the other hand, I do agree that `asnyc` connections deliver much better performance. So if your web-frontend is `async`, use `mobc` and `mobc-arangors`, as it's `async` too. I'm currently working a lot with `Rocket`, and the stable branch currently is synchronous, so I'm sticking with this implementation here until the async version of `Rocket` is stable.

Weasy666 committed:
Update arangors to 0.5 and add uclient (#3)

Uclient is now needed, because arangors uses it as a unified client
interface, but does not re-export it.

Commit:
Add

Commit:
Switch to git for 'arangors' due to compile error in 0.5
